### PR TITLE
Added the format_dict.py module that converts normal csv dictionaries to Netbox friendly dictionaries.

### PR DESCRIPTION
**Pull request by khalford**
Pynetbox friendly dictionaries

### CODE REVIEW

File: Netbox_CSV_Read/CSV/format_dict.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+from typing import Optional
+from Enums.dcim_device_id import DeviceInfoID
+from Enums.dcim_device_no_id import DeviceInfoNoID
+from Netbox_Api.netbox_connect import NetboxConnect
+from operator import attrgetter
+
+
+class FormatDict(NetboxConnect):
```
Comment (meoflynn) [inline Netbox_CSV_Read/CSV/format_dict.py:8]: Could you add a docstring to describe this class?

File: Netbox_CSV_Read/Enums/dcim_device_id.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from enum import Enum
+
+
+class DeviceInfoID(Enum):
```
Comment (meoflynn) [inline Netbox_CSV_Read/Enums/dcim_device_id.py:4]: Could you add a docstring to describe this class?

File: Netbox_CSV_Read/Enums/dcim_device_no_id.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from enum import Enum
+
+
+class DeviceInfoNoID(Enum):
```
Comment (meoflynn) [inline Netbox_CSV_Read/Enums/dcim_device_no_id.py:4]: Could you add a docstring to this class?

**Review by meoflynn — CHANGES_REQUESTED** [review]

File: Netbox_CSV_Read/CSV/csv_utils.py [hunk 4]
```diff
@@ -1,19 +1,21 @@
-import pandas as pd
+import pandas
```
Comment (DavidFair) [inline Netbox_CSV_Read/CSV/csv_utils.py:1]: `import pandas as pd`

File: Netbox_CSV_Read/CSV/csv_utils.py [hunk 5]
```diff
@@ -1,19 +1,21 @@
...
 
 
 class CsvUtils:
     """
     This class provides methods to read data from CSV files and allow the data to be easily read and used elsewhere.
     """
-    @staticmethod
-    def csv_to_python(file_path: str) -> dict:
+    def __init__(self):
+        self.pd = pandas
```
Comment (DavidFair) [inline Netbox_CSV_Read/CSV/csv_utils.py:9]: ...^ then we don't need this

File: Netbox_CSV_Read/CSV/format_dict.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        """
+        if not api:
+            self.netbox = NetboxConnect(url, token).api_object()
+        else:
+            self.netbox = api
+        self.dicts = dicts
+        self.enums_id = DeviceInfoID
+        self.enums_no_id = DeviceInfoNoID
+
+    def iterate_dicts(self) -> list:
```
Comment (DavidFair) [inline Netbox_CSV_Read/CSV/format_dict.py:26]: This should be `List` and `Dict`, as `list != List` (it's one of those Python gotchas that was only fixed in later versions which we partially have access to)

File: Netbox_CSV_Read/CSV/format_dict.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        if attr_string == "DEVICE_TYPE":
+            value = value.get(slug=netbox_value).id
+        elif attr_string == "LOCATION":
+            if type(site_value) == int:
+                site_name = self.netbox.dcim.sites.get(site_value).name
+                site_slug = site_name.replace(" ", "-").lower()
+            value = value.get(name=netbox_value, site=site_slug)
+        else:
+            value = value.get(name=netbox_value).id
+        return value
```
Comment (DavidFair) [inline Netbox_CSV_Read/CSV/format_dict.py:69]: Nit: Format dict should only format dicts

This should be in a different class, such as `NetboxData`

**Review by DavidFair — CHANGES_REQUESTED** [review]

File: Netbox_CSV_Read/CSV/format_dict.py (hunk 6 above)
Comment (khalford) [reply Netbox_CSV_Read/CSV/format_dict.py:28]: If the return typing should be uppercase List and Dict, does this apply to argument typing? like:
def func(values: List):
    pass
or
def func(values: list):
    pass

File: Netbox_CSV_Read/CSV/format_dict.py (hunk 7 above)
Comment (khalford) [reply Netbox_CSV_Read/CSV/format_dict.py]: 👍

**Review by meoflynn — CHANGES_REQUESTED** [review]

**Pull request merged**